pymobiledevice3/services/device_link.py

```diff
@@ -77,20 +77,35 @@ def download_files(self, message):
         for file in message[1]:
             self.service.sendall(struct.pack(SIZE_FORMAT, len(file)))
             self.service.sendall(file.encode())
+            
             try:
-                data = (self.root_path / file).read_bytes()
+                file_path = self.root_path / file
+                file_size = file_path.stat().st_size
+                
+                # split into chunks, otherwise we may crash BackupAgent2 by OOM
+                # https://github.com/doronz88/pymobiledevice3/issues/1165#issuecomment-2376815692
+                chunk_size = 128 * 1024 * 1024 # 128 MB
+                
+                file_handle = file_path.open('rb')
```

Comment:
refactor into a context-manager so the resource is closed as soon as it's not needed any longer
```suggestion
                with file_path.open('rb') as file_handle:
```